ui/src/backoffice/common/components/ResultItem/ResultItem.tsx

```diff
@@ -23,43 +25,30 @@ const renderWorkflowStatus = (status: string) => {
 };
 
 const WorkflowResultItem = ({ item }: { item: any }) => {
+  const workflowId = item?.get('id');
   const data = item?.get('data');
-  const decision = item?.get('decisions')?.first();
+  const acquisitionSourceDatetime = new Date(
+    data?.getIn(['acquisition_source', 'datetime'])
+  ).toLocaleDateString();
+  const acquisitionSourceSource = data?.getIn(['acquisition_source', 'source']);
+  const acquisitionSourceEmail = data?.getIn(['acquisition_source', 'email']);
+  const workflowTypeKey = item?.get(
+    'workflow_type'
+  ) as keyof typeof WORKFLOW_TYPES;
+  const workflowTypeToPidTtype = WORKFLOW_TYPES[workflowTypeKey];
+  const isAuthor = workflowTypeToPidTtype === AUTHORS_PID_TYPE;
 
   return (
     <div className="result-item result-item-action mv2">
       <Row justify="start" wrap={false}>
-
         <Col className="col-details">
           <ResultItem>
             <Link
               className="result-item-title"
-              to={`${BACKOFFICE}/${item.get('id')}`}
+              to={`${BACKOFFICE}/${workflowId}`}
               target="_blank"
             >
-              <div className="flex">
-                <div style={{ marginTop: '-2px' }}>
-                  <UnclickableTag>Author</UnclickableTag>
-                  {item?.get('workflow_type') === 'AUTHOR_UPDATE' && (
-                    <>
-                      {' '}
-                      <UnclickableTag color="processing">Update</UnclickableTag>
-                    </>
-                  )}
-                  {decision && (
-                    <UnclickableTag
-                      className={`decision-pill ${resolveDecision(
-                        decision?.get('action')
-                      )?.bg}`}
-                    >
-                      {resolveDecision(decision?.get('action'))?.text}
-                    </UnclickableTag>
-                  )}
-                </div>
-                <span className="dib ml2">
-                  {data?.getIn(['name', 'value'])}
-                </span>
-              </div>
+              {isAuthor && <AuthorResultItem item={item} />}
```

Comment:
Can we have a factory for that with cases? Because if we add more, we will add more and more lines like this one.